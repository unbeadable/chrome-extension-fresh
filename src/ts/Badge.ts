import { BadgeIcon } from './BadgeIcon';

const badgeText: () => HTMLElement = () => {
    const element = document.createElement('span');
    element.setAttribute('class', 'unbdbl-primary-text');
    element.innerText =
        'This product contributes to the pollution of our oceans.';
    return element;
};

export class Badge {
    readonly element: HTMLElement = document.createElement('div');

    constructor() {
        this.element.setAttribute('class', 'unbdbl-badge unbdbl-primary-text');
        this.element.appendChild(new BadgeIcon().element);
        this.element.appendChild(badgeText());
    }
}
