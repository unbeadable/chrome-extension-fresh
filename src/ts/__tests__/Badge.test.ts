import {Badge} from '../Badge';

describe('Badge', () => {
    it('should match snapshot', () => {
        expect(new Badge().element).toMatchSnapshot();
    });
});