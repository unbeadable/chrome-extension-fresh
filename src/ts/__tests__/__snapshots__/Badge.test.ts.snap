// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should match snapshot 1`] = `
<div
  class="unbdbl-badge unbdbl-primary-text"
>
  <img
    class="unbdbl-badge-icon"
    src="https://s3.eu-central-1.amazonaws.com/unbeadable/baseline-report-24px.svg"
  />
  <span
    class="unbdbl-primary-text"
  />
</div>
`;
